test/unit/stablizer/TribeReserveStabilizer.test.ts

```diff
@@ -21,180 +29,196 @@ describe('TribeReserveStabilizer', function () {
-        await this.reserveStabilizer
-          .connect(impersonatedSigners[governorAddress])
-          .setCollateralizationThreshold('9900', {});
+      it('reset reverts', async function () {
+        await reserveStabilizer.connect(impersonatedSigners[governorAddress]).setCollateralizationThreshold('9900', {});
+        await reserveStabilizer.resetOracleDelayCountdown();
+
         await expectRevert(
-          this.reserveStabilizer.connect(impersonatedSigners[userAddress]).exchangeFei(40000000, {}),
-          'TribeReserveStabilizer: Collateralization ratio above threshold'
+          reserveStabilizer.connect(impersonatedSigners[userAddress]).exchangeFei(40000000, {}),
+          'Timed: time not ended'
         );
       });
     });
 
     describe('Paused', function () {
       it('reverts', async function () {
-        await this.reserveStabilizer.connect(impersonatedSigners[governorAddress]).pause({});
+        await reserveStabilizer.connect(impersonatedSigners[governorAddress]).pause({});
         await expectRevert(
-          this.reserveStabilizer.connect(impersonatedSigners[userAddress]).exchangeFei(toBN('400000'), {}),
+          reserveStabilizer.connect(impersonatedSigners[userAddress]).exchangeFei(toBN('400000'), {}),
           'Pausable: paused'
         );
       });
     });
+    describe('Not Enough FEI', function () {
+      it('reverts', async function () {
+        await expectRevert(
+          reserveStabilizer.connect(impersonatedSigners[userAddress]).exchangeFei(50000000, {}),
+          'ERC20: transfer amount exceeds balance'
+        );
+      });
+    });
   });
 
   describe('isCollateralizationBelowThreshold', function () {
     it('above', async function () {
-      await this.reserveStabilizer
-        .connect(impersonatedSigners[governorAddress])
-        .setCollateralizationThreshold('9900', {});
-      expect(await this.reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(false);
+      await reserveStabilizer.connect(impersonatedSigners[governorAddress]).setCollateralizationThreshold('9900', {});
+      expect(await reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(false);
     });
 
     it('at', async function () {
-      expect(await this.reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(true);
+      expect(await reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(true);
     });
 
     it('below', async function () {
-      await this.reserveStabilizer
-        .connect(impersonatedSigners[governorAddress])
-        .setCollateralizationThreshold('10000', {});
-      expect(await this.reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(true);
+      await reserveStabilizer.connect(impersonatedSigners[governorAddress]).setCollateralizationThreshold('10000', {});
+      expect(await reserveStabilizer.isCollateralizationBelowThreshold()).to.be.equal(true);
     });
 
     it('invalid oracle', async function () {
```

Comment:
use more descriptive test language, ie "should revert when collateralizationOracle reports invalid status"